python/nav/portadmin/snmp/base.py

```diff
@@ -303,15 +303,6 @@ def set_interface_down(self, interface):
             self.IF_ADMIN_STATUS, interface.ifindex, "i", self.IF_ADMIN_STATUS_DOWN
         )
 
-    def cycle_interface(self, interface, wait=5.0):
-        wait = int(wait)
-        self.set_interface_down(interface)
-        _logger.debug('Interface set administratively down - '
-                      'waiting %s seconds', wait)
-        time.sleep(wait)
-        self.set_interface_up(interface)
-        _logger.debug('Interface set administratively up')
-
```

Comment:
So SNMP can use the parent's `cycle_interfaces` unchanged?

Reply:
Yes, the routine is the same regardless of internal implementation details of the protocol, so I extracted this implementation and put it in the base class instead. The Juniper version overrides the parent implementation only to force the commit argument to always be true (it is always necessary on Juniper).